### ISSUE TITLE
Impossible to unignore vendor folder

### ISSUE THREAD

**Issue by philvdb** [JetBrains/idea-gitignore#554]
### Prerequisites

* [X] Plugin is in the latest version
* [X] Issue was not reported yet
* [ ] Stack trace (if provided) contains `mobi.hsz.idea.gitignore` package name

### Description

Plugin kept marking my vendor folder as ignored no matter what I do, only solution was to uninstall the plugin.

### Steps to Reproduce

1. Open a project that uses composer containing a vendor folder
2. Install the .ignore plugin
3. Search for a way to unignore the vendor folder

**Expected behavior:** When I remove the automatically-added directories inside the vendor folder in the settings, I expect the vendor folder to not be ignored anymore. It is not in any other ignore file inside the project.

**Actual behavior:** Even after restart of phpStorm, all files and folders inside the vendor folder remain ignored.

**Reproduces how often:** 100%.

### Versions

**Plugin:** 2.6. something - it's uninstalled now. Was the latest version at the time.

**IDE:** PhpStorm 2018.1.6
Build #PS-181.5281.35, built on June 14, 2018
JRE: 1.8.0_152-release-1136-b39 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.12.6 

**OS:** see above

**Comment by hsz** [comment]
@philvdb Thanks for creating this issue. Can you please provide the `.gitignore` file that you use?

**Comment by philvdb** [comment]
My use case is that I am working in a subfolder of `/vendor` myself and could not find a way to unignore that specific folder in phpStorm, marking my folder as sources root was no help, but when even manually removing all the `/vendor/*` folders from the ignore settings didn't remedy the problem, I created this issue.

**Comment by rvfischione** [comment]
> My use case is that I am working in a subfolder of /vendor myself and could not find a way to unignore that specific folder

@philvdb Sorry if I'm misunderstanding the situation, but it looks like [this SO answer](https://stackoverflow.com/questions/5533050/gitignore-exclude-folder-but-include-specific-subfolder) applies here: if a whole folder is ignored, everything underneath will also be ignored, even if some other pattern allows for a subfolder. Does the following work, if you put it in all of the .gitignore files in your project?

```
!/vendor/

/vendor/*
!/vendor/included_folder
```

**Comment by ethanbeyer** [comment]
I know this comment is super late, and may not apply...but who knows. I found a bug in one of my repos where every single folder called "vendor" was ignored. Unfortunately, some of those "vendor" folders were apart of plugins and needed to be present in deployments.

I changed `vendor` to `/vendor` in my project root `.gitignore`, but could never add the vendor folders in other parts of the tree.

And then I discovered that at some point in my life, I added `vendor` to my `~/.gitignore_global` file.

May just help someone who made the same mistake.
